packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1093,6 +1108,15 @@ export class RemoteFluidDataStoreContext extends FluidDataStoreContext {
 	private readonly initialSnapshotDetailsP = new LazyPromise<ISnapshotDetails>(async () => {
 		// Sequence number of the snapshot.
 		let sequenceNumber: number | undefined;
+		// Check whether we need to fetch the snapshot first to load.
+		if (this.snapshotFetchRequired === undefined && this._baseSnapshot?.groupId !== undefined) {
+			assert(this.blobContents !== undefined, "Blob contents should be present to evaluate");
+			assert(this._baseSnapshot !== undefined, "snapshotTree should be present to evaluate");
+			this.snapshotFetchRequired = isSnapshotFetchRequiredForLoadingGroupId(
+				this._baseSnapshot,
+				this.blobContents,
+			);
+		}
 		if (this.snapshotFetchRequired) {
```

Comment:
It's weird to think that this will always be true even after we've made the snapshotFetch. What do you think?

Reply:
Why would this be true always? I am setting it to false once I fetch the contents.